### ISSUE TITLE
ilastik autocontext 3-stage eats up ram+swap and then get oom killed

### ISSUE THREAD

**Issue by svenpeter42** [ilastik/ilastik#1376]
I have a 5kx5k 2D image and use the three stage autocontext workflow with ~12-15 classes in stage 1 and 2. After adding enough labels in stage one and two ilastik eats up all of my RAM (64GiB) + all of my swap space (another 128 GiB) and then is killed by the kernel once it tries to allocate even more memory.

If I manually limit the RAM size to 16GiB it still eats up at least 100GiB.

**Comment by akreshuk** [comment]
Yeah, well, I don't recommend using 3-stage in one go. In terms of resources, the best way is to train 3 separate pixel classifications, adding the prob maps as channels in between from a script. For those who don't want to do it from a script, you can open a 2-stage autocontext workflow and then, after training only the first stage, use the export applet and choose "input to 2nd stage" as source. Export that and start a new pixel classification project with that as input. 

That said, of course it shouldn't eat RAM above your limit, thanks for reporting it. Are you using the latest from source? At least one big memory problem got fixed in the upcoming release, but more might be lurking in there. Test binaries for the release are available on request.

**Comment by svenpeter42** [comment]
I can try the latest source release later this weeks. Right now I've just moved to my own scripts that don't use any of ilastik codebase.
